drivers/gpio/gpio_sifive.c

```diff
@@ -117,25 +117,34 @@ static int gpio_sifive_config(struct device *dev,
 	}
 
```

Comment:
The driver should return `-ENOTSUP` if an open source / open drain configuration is requested.
```
	if ((flags & GPIO_SINGLE_ENDED) != 0) {
		return -ENOTSUP;
	}
```